app/src/androidTest/java/org/mozilla/fenix/ui/SettingsSearchTest.kt

```diff
@@ -370,4 +374,114 @@ class SettingsSearchTest {
+            clickSearchEnginePrompt(activityTestRule, "DuckDuckGo")
+        }.submitQuery("firefox") {
+            verifyUrl("duckduckgo.com/?q=firefox")
+        }
+    }
+
+    // Expected for app language set to Arabic
+    @Test
+    fun verifySearchEnginesWithRTLLocale() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            toggleShowSearchShortcuts()
+        }.goBack {
+        }.openLanguageSubMenu {
+            TestHelper.registerAndCleanupIdlingResources(
+                RecyclerViewIdlingResource(
+                    activityTestRule.activity.findViewById(R.id.locale_list),
+                    2,
+                ),
+            ) {
+                selectLanguage("Arabic")
+                verifyLanguageHeaderIsTranslated(ARABIC_LANGUAGE_HEADER)
+            }
+        }
+
+        exitMenu()
+
+        homeScreen {
+        }.openSearch {
+            verifyTranslatedFocusedNavigationToolbar("ابحث أو أدخِل عنوانا")
+            verifySearchEngineShortcuts(
+                activityTestRule,
+                "Google",
+                "Bing",
+                "Amazon.com",
+                "DuckDuckGo",
+                "eBay",
+                /* Disabled Arabic Wikipedia verification
+                   until https://github.com/mozilla-mobile/fenix/issues/12236 gets fixed
+                 "ويكيبيديا (ar)"
+                */
+            )
+        }
+    }
+
+    // Expected for en-us defaults
+    @Test
+    fun toggleSearchEnginesShortcutListTest() {
+        homeScreen {
+        }.openThreeDotMenu {
+        }.openSettings {
+        }.openSearchSubMenu {
+            verifyShowSearchEnginesToggleState(false)
+            toggleShowSearchShortcuts()
+            verifyShowSearchEnginesToggleState(true)
+        }
+
+        exitMenu()
+
+        homeScreen {
+        }.openSearch {
+            verifySearchEngineShortcuts(
+                activityTestRule,
+                "Google",
+                "Bing",
+                "Amazon.com",
+                "DuckDuckGo",
+                "eBay",
+                "Wikipedia",
+            )
+        }.clickSearchEngineSettings(activityTestRule) {
+            toggleShowSearchShortcuts()
+            verifyShowSearchEnginesToggleState(false)
+        }
+
+        exitMenu()
+
+        homeScreen {
+        }.openSearch {
+            verifySearchEngineShortcutsAreNotDisplayed(
+                activityTestRule,
+                "Google",
+                "Bing",
+                "Amazon.com",
+                "DuckDuckGo",
+                "eBay",
+                "Wikipedia",
+            )
+            clickSearchEngineShortcutButton()
+            verifySearchEngineShortcuts(
+                activityTestRule,
+                "Google",
+                "Bing",
+                "Amazon.com",
+                "DuckDuckGo",
+                "eBay",
+                "Wikipedia",
+            )
+        }
```

Comment:
I think I asked before a while ago but I forget, is it possible to query these defaults (and store them in a list once) from the search component? https://github.com/mozilla-mobile/firefox-android/tree/main/android-components/components/feature/search

Reply:
I don't recall, but I can give it a try.
I'll try in a separate PR, and if it works, I'll update the corresponding tests afterwards.